### ISSUE TITLE
The EngineV2Configuration function is very slow for large schema

### ISSUE THREAD

**Issue by dnerdy** [jensneuse/graphql-go-tools#294]
Our service schemas are 1k-4k lines each, and producing an engine configuration for all services (around 35k lines total) takes about 15-20 seconds on a 2.7 GHz Quad-Core Intel Core i7.

The bottleneck appears to be the `baseSchema` method on `LocalTypeFieldExtractor` which prints and parses the schemas multiple times for each type. Engine construction time is important to us because we construct multiple engines and replace engines as schemas updated.

#293 is also related to the federation config factory. It could make sense to address both issues at the same time since node traversal needs to be updated for #293.

Would you be up for us optimizing construction of the engine config?

**Comment by jensneuse** [comment]
@dnerdy How long does it take with your updated branch with the improvements? Is the processing time now acceptable?

**Comment by dnerdy** [comment]
Yes! Constructing an engine configuration now takes around 60ms for our schema instead of 15-20 seconds, which is great.
